src/stream/pipeline/runner.rs

```diff
@@ -26,8 +27,11 @@ impl PipelineRunner {
         let pipeline_weak = pipeline.downgrade();
         let (killswitch_sender, _killswitch_receiver) = broadcast::channel(1);
         let watcher_killswitch_receiver = killswitch_sender.subscribe();
+        let (start_signal_sender, start_signal_receiver) = broadcast::channel(1);
```

Comment:
just to point out that while solutions using the `oneshot` channel or `barrier` would have solved it, the `broadcast` channel is the easiest to deal with without fighting with async.